### PR TITLE
Always write cardinality to function.json

### PR DESCRIPTION
**Pull request by Flanker32**
Always write cardinality to function.json, for issue #561

### CODE REVIEW

**Comment by jdneo** [comment]
commit message typo: Alawys

File: azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/bindings/Binding.java [hunk 1]
```diff
@@ -24,6 +26,13 @@
 
     protected Map<String, Object> bindingAttributes = new HashMap<>();
 
+    protected static Map<BindingEnum, List<String>> requiredAttributeMap = new HashMap<>();
+
+    static {
+        //initialize required attributes, which will save to function.json even if it equals to its default value
```
Comment (andxu) [inline azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/bindings/Binding.java]: be saved

**Comment by Flanker32** [comment]
@pragnagopa  Test case has been added, one question is maven plugin will read the property of annotations and write it to function.json directly, and the value will be like `"cardinality" : "MANY"`, will it work as well ?

**Comment by pragnagopa** [comment]
Yes. "cardinality" : "MANY" works

**Pull request merged**